src/weights.jl

```diff
@@ -337,8 +364,12 @@ end
 
 for w in (AnalyticWeights, FrequencyWeights, ProbabilityWeights, Weights)
     @eval begin
-        Base.isequal(x::$w, y::$w) = isequal(x.sum, y.sum) && isequal(x.values, y.values)
-        Base.:(==)(x::$w, y::$w)   = (x.sum == y.sum) && (x.values == y.values)
+        Base.isequal(x::$w, y::$w) =
+            (x.values isa SubArray || y.values isa SubArray || isequal(x.values, y.values)) &&
```

Comment:
shouldn't `isequal(x.values, y.values)` be `sum(x) == sum(y)` here?

Reply:
Woops. Good catch. Though this is one of the few places where I think accessing the field directly is clearer, as we certainly don't want `sum(x)` to recompute the sum given that it's just an optimization to return `false` quickly if sums are different.
```suggestion
            (x.values isa SubArray || y.values isa SubArray || isequal(x.sum, y.sum)) &&
```

Reply:
ok